DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h

```diff
@@ -0,0 +1,97 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_MCCOMPLABEL_H
+#define ALICEO2_MCCOMPLABEL_H
+
+#include <Rtypes.h>
+#include <iomanip>
+#include <ios>
+#include <iostream>
+
+namespace o2
+{
+// Composed Label to encode MC track id, event it comes from and the source (file)
+
+class MCCompLabel
+{
+ private:
+  static constexpr ULong64_t ul0x1 = 0x1;
+  static constexpr ULong64_t NotSet = 0xffffffffffffffff;
+
+  ULong64_t mLabel = NotSet; ///< MC label encoding MCtrack ID and MCevent origin
+
+ public:
+
+  // number of bits reserved for MC track ID, DON'T modify this, sine the
+  // track ID might be negative
+  static constexpr int nbitsTrackID = 32;
+  static constexpr int nbitsEvID = 19;    // number of bits reserved for MC event ID
+  static constexpr int nbitsSrcID = 8;    // number of bits reserved for MC source ID
```

Comment:
idea: we could potentially try to make MCCompLabel configurable by passing these things as template parameters?